ifrs17/Import/Importers.ipynb

```diff
@@ -673,44 +718,64 @@
                 "\n",
-                "\n    var committedYieldCurves = await DataSource.Query<YieldCurve>().ToArrayAsync();",
+                "\n    var committedYieldCurves = await options.TargetDataSource.Query<YieldCurve>().ToArrayAsync();",
                 "\n    var hasNameColumn = dataSet.Tables[ImportFormats.YieldCurve].Columns.Any(x => x.ColumnName == nameof(YieldCurve.Name));",
-                "\n    var importLog = await Import.FromDataSet(dataSet).WithType<YieldCurve>((dataset, datarow) =>  { ",
-                "\n                var yieldCurve = new YieldCurve {",
-                "\n                Currency = datarow.Field<string>(nameof(YieldCurve.Currency)),",
-                "\n                Year = args.Year,",
-                "\n                Month = args.Month, ",
-                "\n                Scenario = args.Scenario,",
-                "\n                Values = datarow.Table.Columns.Where(c => c.ColumnName.StartsWith(nameof(YieldCurve.Values))).OrderBy(c => c.ColumnName.Length).ThenBy(c => c.ColumnName)",
-                "\n                                .Select(x => datarow.Field<string>(x.ColumnName).CheckStringForExponentialAndConvertToDouble()).ToArray(),",
-                "\n                Name = hasNameColumn",
-                "\n                            ? datarow.Field<string>(nameof(YieldCurve.Name))",
-                "\n                            : default(string)",
-                "\n            };",
-                "\n            return committedYieldCurves.Contains(yieldCurve, YieldCurveComparer.Instance())",
-                "\n                    ? null",
-                "\n                    : yieldCurve;",
-                "\n            }",
-                "\n        ).WithTarget(workspace).ExecuteAsync(); ",
-                "\n    ",
+                "\n    var importLog = await Import.FromDataSet(dataSet).WithType<YieldCurve>((dataset, datarow) => { ",
+                "\n        var yieldCurve = new YieldCurve { ",
+                "\n            Currency = datarow.Field<string>(nameof(YieldCurve.Currency)),",
+                "\n            Year = primaryArgs.Year,",
+                "\n            Month = primaryArgs.Month, ",
+                "\n            Scenario = primaryArgs.Scenario,",
+                "\n            Values = datarow.Table.Columns.Where(c => c.ColumnName.StartsWith(nameof(YieldCurve.Values))).OrderBy(c => c.ColumnName.Length).ThenBy(c => c.ColumnName)",
+                "\n                            .Select(x => datarow.Field<string>(x.ColumnName).CheckStringForExponentialAndConvertToDouble()).ToArray(),",
+                "\n            Name = hasNameColumn ? datarow.Field<string>(nameof(YieldCurve.Name)) : default(string)",
+                "\n        };",
+                "\n        return committedYieldCurves.Contains(yieldCurve, YieldCurveComparer.Instance()) ? null : yieldCurve;",
```

Comment:
I think this yield curve comparer needs improvement. Do we have an issue for that? @calderinisystemorph 
For example importing the same YC for 2 different months is not required and then a new YC should be stored. 
@amuolo can you add this issue to the board? thanks

Reply:
done #191